### PR TITLE
Custom AoI upload

### PR DESCRIPTION
**Pull request by nerik**
Migrates the previous upload custom AoI modal to the Exploration page.

Screenshot 2023-10-17 at 15 12 35

### CODE REVIEW

File: app/scripts/components/common/map/controls/aoi/custom-aoi-control.tsx [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+function CustomAoI({ onConfirm }: CustomAoIProps) {
+  const [aoiModalRevealed, setAoIModalRevealed] = useState(false);
+  return (
+    <>
+      <SelectorButton>
+        <CollecticonUpload2
+          title='Upload geoJSON'
+          meaningful
+          onClick={() => setAoIModalRevealed(true)}
+        />
```
Comment (danielfdsilva) [inline app/scripts/components/common/map/controls/aoi/custom-aoi-control.tsx:36]: The click listener should be on the button
```suggestion
      <SelectorButton onClick={() => setAoIModalRevealed(true)}>
        <CollecticonUpload2 title='Upload geoJSON' meaningful />
```

**Review by danielfdsilva — CHANGES_REQUESTED** [review]
Maybe not part of this PR, but could be good to allow the user to clear all the AOIs by pressing the trash can with nothing selected.

**Comment by nerik** [comment]
@danielfdsilva Good suggestion. I don't see any other way to implement this (ie adding a custom listener on the trash button) than to rebuild the control from scratch, thoughts?

**Comment by danielfdsilva** [comment]
Yeah... seems to be that way. I've rarely used the default controls. For the current analysis I had [created custom ones](https://github.com/NASA-IMPACT/veda-ui/blob/main/app/scripts/components/analysis/define/aoi-selector.tsx) and then [handled the events](https://github.com/NASA-IMPACT/veda-ui/blob/main/app/scripts/components/common/aoi/use-aoi-controls.ts) as I needed.

**Comment by danielfdsilva** [comment]
There could be also an additional button to do that, maybe it is also clearer? like a reset?
The current analysis uses the arrows one:
image

**Pull request merged**